cfg_mgmt/gcp.py

```diff
@@ -53,10 +54,19 @@ def delete_service_account_key(
 
 
 def rotate_cfg_element(
-    cfg_element: model.container_registry.ContainerRegistryConfig,
+    cfg_element: model.container_registry.ContainerRegistryConfig | model.gcp.GcpServiceAccount,
     cfg_factory: model.ConfigFactory,
 ) ->  typing.Tuple[cfg_mgmt.revert_function, dict, model.NamedModelElement]:
-    client_email = json.loads(cfg_element.password())['client_email']
+    if not type(cfg_element) in (
+        model.container_registry.ContainerRegistryConfig,
+        model.gcp.GcpServiceAccount,
+    ):
+        raise TypeError(cfg_element)
+
+    if isinstance(cfg_element, model.container_registry.ContainerRegistryConfig):
+        client_email = json.loads(cfg_element.password())['client_email']
```

Comment:
it would be nicer, and more consistent to remaining codebase to implement the retrieval of client_email and private_key_id in model-classes at `model` package

Reply:
Agree, [adjusted ContainerRegistryConfig](https://github.com/gardener/cc-utils/commit/be471e4fd08402cf01a292d66878b0d4f1a500df) and will also do for GcpServiceAccount.